itou/asp/tests.py

```diff
@@ -0,0 +1,181 @@
+
+        user = _users_with_mock_address(10)
+        result, error = format_address(user, strict=False)
+        self.assertIsNone(error)
+        self.assertEquals(result.get("lane_type"), LaneType.SQ.name)
+        self.assertEquals(result.get("number"), "2")
+
+        user = _users_with_mock_address(11)
+        result, error = format_address(user, strict=False)
+        self.assertIsNone(error)
+        self.assertEquals(result.get("lane_type"), LaneType.HAM.name)
+
+        user = _users_with_mock_address(12)
+        result, error = format_address(user, strict=False)
+        self.assertIsNone(error)
+        self.assertEquals(result.get("lane_type"), LaneType.QUAR.name)
+        self.assertEquals(result.get("number"), "16")
+
+        user = _users_with_mock_address(13)
+        result, error = format_address(user, strict=False)
+        self.assertIsNone(error)
+        self.assertEquals(result.get("lane_type"), LaneType.RES.name)
+        self.assertEquals(result.get("number"), "1")
+
+        user = _users_with_mock_address(14)
+        result, error = format_address(user, strict=False)
+        self.assertIsNone(error)
+        self.assertEquals(result.get("lane_type"), LaneType.VOIE.name)
+        self.assertEquals(result.get("number"), "172")
+
+        user = _users_with_mock_address(15)
+        result, error = format_address(user, strict=False)
+        self.assertIsNone(error)
+        self.assertEquals(result.get("lane_type"), LaneType.ALL.name)
+        self.assertEquals(result.get("number"), "3")
+
+
+class LaneTypeTest(TestCase):
+    def test_aliases(self):
+        "Test some variants / alternatives used by api.geo.gouv.fr for lane types"
+        self.assertEquals(LaneType.GR, find_lane_type_aliases("grand rue"))
+        self.assertEquals(LaneType.GR, find_lane_type_aliases("grande-rue"))
+        self.assertEquals(LaneType.RUE, find_lane_type_aliases("R"))
+        self.assertEquals(LaneType.RUE, find_lane_type_aliases("r"))
+        self.assertEquals(LaneType.LD, find_lane_type_aliases("lieu dit"))
+        self.assertIsNone(find_lane_type_aliases("XXX"))
+
+
+class LaneExtensionTest(TestCase):
+    @mock.patch(
+        "itou.utils.address.format.get_geocoding_data", side_effect=mock_get_geocoding_data,
+    )
+    def test_standard_extension(self, _):
+        "Check if lane extension is included in ASP ref file"
+        user = _users_with_mock_address(0)
+        result, error = format_address(user, strict=False)
+        self.assertEquals(result.get("std_extension"), LaneExtension.B.name)
+
+        user = _users_with_mock_address(16)
+        result, error = format_address(user, strict=False)
+        self.assertEquals(result.get("std_extension"), LaneExtension.T.name)
+
+    @mock.patch(
+        "itou.utils.address.format.get_geocoding_data", side_effect=mock_get_geocoding_data,
+    )
+    def test_non_standard_extension(self, _):
+        "Non-standard extension, i.e. not in ASP ref file"
```

Comment:
Attention tu as mis une string là.... je pense que tu voulais un one line docstring :

`"""Non-standard extension, i.e. not in ASP ref file"""`